src/d3a/models/state.py

```diff
@@ -50,6 +50,9 @@
 class StateInterface(ABC):
     """Interface containing methods that need to be defined by each State class."""
 
+    def __init__(self):
+        self._spot_market_time_slot: DateTime = None
```

Comment:
I still vote for not adding any attribute to the state interface, we can still stick to the old function headers where we pass the market slot

Reply:
Here I would like to tag @spyrostz as I added it after discussing with him.